### DISCUSSION TITLE
What is the correct syntax to add  a document library(s) to the Query Template along with my RefinableStrings?

### DISCUSSION THREAD

**Discussion by tedtninety** [microsoft-search/pnp-modern-search#2860] [Q&A] [answered]
Hello,

I have several RefinableStrings in my Query Template which is a metadata from a single list.
I was attempting to add a document library, and perhaps another list to the search terms  ( assuming this is possible, and following the correct procedure.) Would you know what the correct syntax and way to execute this  correctly please?

`RefinableString100:({SearchTerms}) OR
RefinableString101:({SearchTerms}) OR
RefinableString102:({SearchTerms}) OR
RefinableString90:({SearchTerms})`

I tried adding a path to a document library :
path:xttps://tenant.sharepoint.com/sites/mysite/Shared%20Documents/Forms/Allitems.aspx

However, the result is a blank page with no results at all, including nothing from the RefinableStrings which produced results prior to adding the aforementioned path.

Thank you again for any assistance.
ted

**Comment by kasperbolarsen** [comment, marked as answer]
The path part should be:
path:https://tenant.sharepoint.com/sites/mysite/Shared%20Documents* isDocument:1


As I read the query you only wants a hit when one or more of the Refinablestrings contains the search term, not if the document contains the search term?

**Reply by kasperbolarsen** [reply]
The isDocument is only relevant on Document libraries,  not if you also want to include the list in the scope.

**Reply by tedtninety** [reply]
Hi @kasperbolarsen ,
Thank you for your reply.

What I am trying to accomplish is to keep the RefinableStrings as SearchTerms but I also wanted to add anything in the document library . So all of them together. 
I am assuming this is possible because before you taught me how to use RefinableStrings as a SearchTerm ( and thank you again :) ),I only used paths. I am hoping that I can use SearchTerms and a path.

Is it even possible; to add more than one path?

Thank you so much @kasperbolarsen , please forgive if I do not get to respond quickly, it is now Supper time :)

ted

**Reply by kasperbolarsen** [reply]
No problem in using two or more paths:
(path:https://tenant.sharepoint.com/sites/mysite/Shared%20Documents* OR path:https://tenant.sharepoint.com/sites/Someothersite/Documents*)
will work fine.

I will suggest that you don't use the RefinableStrings in the query if content from the document should be included in the search. Try something like this to start with:

{SearchTerms}
(path:https://tenant.sharepoint.com/sites/mysite/Shared%20Documents* OR Path:https://tenant.sharepoint.com/sites/Someothersite/Documents*)

**Reply by tedtninety** [reply]
Thank you so much @kasperbolarsen! :)

As for the RefinableStrings , the reason I use them is to make certain that a "second title column" and "the description" column can be searched by keyword.
Would I lose the aforementioned if I removed the SeacrhTerms as you had mentioned?
Since I have about 5 "SearchTerms (RefinableStrings) in the Query Template for 5 columns in that list , would it be just as effective to use a path to the list itself instead?
If so?  So that I may understand, when would it be applicable to just use a few SearchTerm(RefinableStrings) as opposed to a path to a whole list?

Thank you again,
ted
